tests/ui/associated-types/dont-suggest-cyclic-constraint.stderr

```diff
@@ -4,8 +4,8 @@ error[E0308]: mismatched types
 LL |     debug_assert_eq!(iter.next(), Some(value));
    |                                   ^^^^^^^^^^^ expected `Option<<I as Iterator>::Item>`, found `Option<&<I as Iterator>::Item>`
```

Comment:
It seems like maybe we should use the expected/found ones now instead for the label, as they end up being significantly shorter.